File: src/main/java/nl/martijnklene/api/infrastructure/model/swagger/LegacyBlogPost.java
package nl.martijnklene.api.infrastructure.model.swagger;

import java.util.Date;

public class LegacyBlogPost extends BlogPayload {
    private Date createdAt;
    private Date publishedAt;

    public Date getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(Date createdAt) {
        this.createdAt = createdAt;
    }

    public Date getPublishedAt() {
        return publishedAt;
    }

    public void setPublishedAt(Date publishedAt) {
        this.publishedAt = publishedAt;
    }
}

File: src/main/kotlin/nl/martijnklene/api/infrastructure/http/model/LegacyBlogPost.kt
package nl.martijnklene.api.infrastructure.http.model

import java.time.ZonedDateTime
import javax.validation.constraints.NotEmpty

data class LegacyBlogPost(
    @NotEmpty val title: String,
    @NotEmpty val content: String,
    @NotEmpty val tags: String,
    @NotEmpty val createdAt: ZonedDateTime,
    @NotEmpty val publishedAt: ZonedDateTime
)
